reporting/reporter.go

```diff
@@ -0,0 +1,274 @@
+package reporting
+
+import (
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	metrics "github.com/rcrowley/go-metrics"
+	"github.com/wavefronthq/wavefront-sdk-go/application"
+	"github.com/wavefronthq/wavefront-sdk-go/histogram"
+	wf "github.com/wavefronthq/wavefront-sdk-go/senders"
+)
+
+// WavefrontMetricsReporter report go-metrics to wavefront
+type WavefrontMetricsReporter interface {
+	Start()
+	Stop()
+	Close()
```

Comment:
Let's just have one Stop() and ensure on stop that we both stop reporting and exit the go routine. Let's also close the sender on Stop.